### DISCUSSION TITLE
Gas Meter monitor using reed contact

### DISCUSSION THREAD

**Discussion by Gamadril** [pvvx/ATC_MiThermometer#364] [Ideas]
Hi,
I would like to use the custom firmware together with a reed switch to monitor the gas meter. Currently the firmware can count the pulses of the reed switch. However this is only useful if you know the current value of the gas meter. My idea is modify the firmware to be able to "config" the current value manually, so the counter would update the real current value internally and send it either once a minute or on demand after connection - not sure what's better for battery life.

I have some experience with embedded systems programming and with some hints I will try to do the modification.
Thanks

**Comment by fegyosz** [comment]
This is the way I am using it to monitor gas and electricity. I am adding the bias in home assistant to show the exact meter values. It works fine until I need to change the battery...
I would also love to see an option to preset the counter value. This would make life much more easy.
Actually, I have visited the PVVX site if there is a solution for this.

**Comment by pvvx** [comment]
[TelinkMiFlasher](https://pvvx.github.io/ATC_MiThermometer/TelinkMiFlasher.html)

![image](https://github.com/user-attachments/assets/7bcd8ef4-48b6-4997-83f8-bb02767199ff)

- Pulse counter bit depth - 32 bits
- The counter counts pulses from zero from power on.
- "Report interval" is the period in seconds between counter transmissions.
- When the pulse counter passes the 16-bit threshold, an additional counter transmission occurs.
- The counter is transmitted in a series of 5 BLE advertisements with an interval of 50 ms.

Warning: ESP32-based receivers miss many BLE packets due to antenna switching to WiFi.
This makes the counter reception very unstable on ESP32-based systems.
